ASTAnalyses/ASTInfo/DeclParser/DeclParser.py

```diff
@@ -61,13 +61,23 @@ def _parse_function_decl(self, function_decl: str) -> Specifier:
             # e.g., 'int ()' is a function that returns int with no parameters
 
             # e.g., 'void (int, int, char *)' is a function that returns void with
-            # 3 parameters (in the order as it appears in C code), int, int, and char *
+            # 3 parameters (in the order as it appears in C code), int, int, and char*
 
             return_type += cur_char
 
             start_index += 1
             cur_char = function_decl[start_index]
 
+        # if return type looks like "int ", change it to "int"
+        if return_type[len(return_type) - 1] == " ":
+            return_type = return_type[:len(return_type) - 1]
+
+        # if return type looks like "int *", change it to "int*"
+        potential_space_char_index = return_type.find("*") - 1
+        if return_type[len(return_type) - 1] == "*" and return_type[potential_space_char_index] == " ":
```

Comment:
Should this be a while loop instead? What if the return type is `"char * * "`?

Reply:
I have yet to see the AST make the return type something like "char * *" but I have seen "char **" or "struct my_struct ****; i.e., the asterisks seem to be grouped together. So I'll add a comment noting that if it looks like "int **", change it to "int**", and that we assume asterisks are grouped together